Gems/ROS2/Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -57,6 +58,23 @@ namespace ROS2
             // Found the component!
             return component;
         }
+
+        //! Returns if entity has component of given type
+        //! @param entity pointer to entity
+        //! @param type tpye of the component
+        //! @returns true if entity has component with given type
+        bool CheckIfEntityHasComponentOfType(const AZ::Entity* entity, const AZ::Uuid type)
```

Comment:
This looks like a candidate for EntityUtils.h 

There is another function to consider here:
        template<class ComponentType>
        inline AZStd::vector<ComponentType*> FindDerivedComponents(Entity* entity)

Reply:
Thanks for tip! Used FindDerivedComponents.